next-frontend/src/lib/staticData/wca-data.ts

```diff
@@ -0,0 +1,177 @@
+import continentDataRaw from "./continents.json";
+import eventsDataRaw from "./events.json";
+import formatsDataRaw from "./formats.json";
+import roundTypesDataRaw from "./round_types.json";
+
+type Country = {
+  info: string;
+  name: string;
+  continentId: string;
+  class: string;
+  iso2: string;
+  id: string;
+};
+
+type Continent = {
+  id: string;
+  name: string;
+  recordName: string;
+  latitude: number;
+  longitude: number;
+  zoom: number;
+};
+
+type Format = {
+  id: string;
+  sortBy: string;
+  sortBySecond: string;
+  expectedSolveCount: number;
+  trimFastestN: number;
+  trimSowestN: number;
+  name: string;
+  shortName: string;
+  allowedFirstPhaseFormats: string[];
+};
+
+type Event = {
+  id: string;
+  rank: number;
+  format: string;
+  name: string;
+  canChangeTimeLimit: boolean;
+  canHaveCutoff: boolean;
+  isTimedEvent: boolean;
+  isFewestMoves: boolean;
+  isMultipleBlindfolded: boolean;
+  isOfficial: boolean;
+  formatIds: string[];
+};
+
+type RoundType = {
+  id: string;
+  rank: number;
+  name: string;
+  cellName: string;
+  final: boolean;
+};
+
+type CamelizedKeys<T> = {
+  [K in keyof T as K extends string ? CamelCase<K> : K]: T[K];
+};
+
+type CamelCase<S extends string> = S extends `${infer Head}_${infer Tail}`
+  ? `${Lowercase<Head>}${Capitalize<CamelCase<Tail>>}`
+  : Lowercase<S>;
+
+function camelizeKeys<T extends Record<string, unknown>>(
+  obj: T,
+): CamelizedKeys<T> {
+  return _.mapKeys(obj, (v, k) => _.camelCase(k)) as CamelizedKeys<T>;
+}
+
+function loadStaticData<T extends Record<string, unknown>>(
+  rawEntities: T[],
+): CamelizedKeys<T>[] {
+  return rawEntities.map(camelizeKeys);
+}
+
+// ----- COUNTRIES -----
+const fictionalCountryIds = _.map(countriesFictive, "id");
+
+const countryData = loadStaticData(
+  // @ts-expect-error currently the fictive countries don't have the same properties as the real countries so ts isn't happy
+  countriesReal.states_lists[0].states.concat(countriesFictive),
+);
+const realCountryData = countryData.filter(
+  (c) => !fictionalCountryIds.includes(c.id),
+);
+
+export const countries = {
+  byIso2: _.mapValues(_.keyBy(countryData, "iso2"), extendCountries),
+  byId: _.mapValues(_.keyBy(countryData, "id"), extendCountries),
+  real: _.map(realCountryData, extendCountries),
+};
+
+function extendCountries(country: Country) {
+  return {
+    ...country,
+    name: (t: (path: string) => string) => t(`countries.${country.iso2}`),
+  };
+}
+
+// ----- CONTINENTS -----
+
+const continentData = loadStaticData(continentDataRaw);
+
+const fictionalContinentIds = ["_Multiple Continents"];
+const realContinents = continentData.filter(
+  (c) => !fictionalContinentIds.includes(c.id),
+);
+
+export const continents = {
+  real: _.map(realContinents, extendContinents),
+};
+
+function extendContinents(continent: Continent) {
+  return {
+    ...continent,
+    name: (t: (path: string) => string) => t(`continents.${continent.name}`),
```

Comment:
I don't think we should keep the names as part of the standard interface

Reply:
cool, that makes it easier. I had to make it a method because i18n now works different in the server or the client